scripts/install_python_dev_dependencies.py

```diff
@@ -0,0 +1,137 @@
+
+    No error is raised if we are running on GitHub Actions because a
+    virtual environment is unnecessary there.
+    """
+    if 'GITHUB_ACTION' in os.environ:
+        return
+    # There are two signals that a virtual environment is active:
+    # * When sys.prefix != sys.base_prefix
+    # * When sys.real_prefix exists
+    # If either is true, we are in a virtual environment. We also check that
+    # sys.real_prefix is Truthy to make testing easier.
+    if (
+        sys.prefix == sys.base_prefix
+        and not (hasattr(sys, 'real_prefix') and getattr(sys, 'real_prefix'))
+    ):
+        raise AssertionError(
+            'Oppia must be developed within a virtual environment.')
+
+
+def install_installation_tools() -> None:
+    """Install the minimal tooling needed to install dependencies."""
+    for package, version in INSTALLATION_TOOL_VERSIONS.items():
+        # We run pip as a subprocess because importing from the pip
+        # module is not supported:
+        # https://pip.pypa.io/en/stable/user_guide/#using-pip-from-your-program.
+        subprocess.run(
+            [sys.executable, '-m', 'pip', 'install', f'{package}=={version}'],
+            check=True,
+            encoding='utf-8',
+        )
+
+
+def install_dev_dependencies() -> None:
+    """Install dev dependencies from
+    COMPILED_REQUIREMENTS_DEV_FILE_PATH.
+    """
+    subprocess.run(
+        ['pip-sync', COMPILED_REQUIREMENTS_DEV_FILE_PATH],
+        check=True,
+        encoding='utf-8',
+    )
+
+
+def compile_pip_requirements(
+    requirements_path: str, compiled_path: str
+) -> bool:
+    """Compile a requirements.txt file.
+
+    Args:
+        requirements_path: str. Path to the requirements.in file.
+        compiled_path: str. Path to the requirements.txt file.
+
+    Returns:
+        bool. Whether the compiled dev requirements file was changed.
+    """
+    with open(compiled_path, 'r', encoding='utf-8') as f:
+        old_compiled = f.read()
+    subprocess.run(
+        [
+            'pip-compile', '--no-emit-index-url', requirements_path,
+            '--output-file', compiled_path,
+        ],
+        check=True,
+        encoding='utf-8',
+    )
+    with open(compiled_path, 'r', encoding='utf-8') as f:
+        new_compiled = f.read()
+
+    return old_compiled != new_compiled
+
+
+def main(cli_args: Optional[List[str]] = None) -> None:
+    """Install all dev dependencies."""
+    args = _PARSER.parse_args(cli_args)
+    assert_in_venv()
+    install_installation_tools()
+    not_compiled = compile_pip_requirements(
+        REQUIREMENTS_DEV_FILE_PATH, COMPILED_REQUIREMENTS_DEV_FILE_PATH)
+    install_dev_dependencies()
+    if args.assert_compiled and not_compiled:
+        raise RuntimeError(
+            'The Python development requirements file '
+            f'{COMPILED_REQUIREMENTS_DEV_FILE_PATH} was changed by the '
+            'installation script. Please commit the changes. '
```

Comment:
Optional: maybe better to say "Please commit the changes using `git commit`"? Not sure.

Reply:
I think it's better to leave this as-is since not everyone uses `git` from the command line (e.g. some people use git through vscode)

Reply:
OK sgtm, thanks.